### PR TITLE
feat: add Danger action.

### PR DESCRIPTION
**Pull request by lucas-zimerman**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [X] Enhancement
- [ ] Refactoring

## :scroll: Description

A new [Github action](https://github.com/danger/danger-js) that throws errors if the changelog wasn't updated (can be skipped by typing #skip-changelog ).
It also gives a tip for new features to be documented at Sentry-docs once they are accepted.

## :bulb: Motivation and Context

Enhanced quality on pull requests.

No open issues.

## :green_heart: How did you test it?
It's just an action, it's quite similar to this one https://github.com/getsentry/sentry-dotnet/pull/625 , also this PR is proof that it works just fine.

## :pencil: Checklist
 
- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] All tests passing
- [ ] No breaking changes

### CODE REVIEW

**Comment by lucas-zimerman** [comment]
@bruno-garcia 
`Found only messages, passing those to review.
Request failed [403]: https://api.github.com/repos/getsentry/sentry-react-native/issues/1216/comments
Feedback: undefined
Response: {
  "message": "Resource not accessible by integration",
  "documentation_url": "https://docs.github.com/rest/reference/issues#create-an-issue-comment"
}
`

I guess it needs something enabled on the project.

File: .github/workflows/danger.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+jobs:
+  build:
+    name: Changelogs
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+      - run: npx danger@10.5.2 ci
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (bruno-garcia) [inline .github/workflows/danger.yml:15]: Same review ;)

```suggestion
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          
```

**Review by bruno-garcia — APPROVED** [review]

**Comment by bruno-garcia** [comment]
> @bruno-garcia
> `Found only messages, passing those to review. Request failed [403]: https://api.github.com/repos/getsentry/sentry-react-native/issues/1216/comments Feedback: undefined Response: { "message": "Resource not accessible by integration", "documentation_url": "https://docs.github.com/rest/reference/issues#create-an-issue-comment" } `
> 
> I guess it needs something enabled on the project.

Not sure what's up here. I can't find any integration I'd need for this.
I'd say we merge and figure out later, might be because you're an outside contributor? That's the reason the E2E builds fail on external PRs.

@HazAT @jennmueng This was taken from [symbolicator](https://github.com/getsentry/symbolicator). It's a good alternative to our current approach (in sentry-java, sentry-dotnet, other) where we have a check that fails the build if there's no changelog without a way to opt-out (like PRs that don't need a changelog).

File: dangerfile.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+function getChangelogDetails() {
+  return `
+<details>
+<summary><b>Instructions and example for changelog</b></summary>
+
+Please add an entry to \`CHANGELOG.md\` to the "Unreleased" section under the following heading:
+ 1. **Feat**: For new user-visible functionality.
+ 2. **Fix**: For user-visible bug fixes.
+ 3. **Ref**: For features, refactory and bug fixes in internal operation.
```
Comment (jennmueng) [inline dangerfile.js:18]:
```suggestion
 1. **feat**: For new user-visible functionality.
 2. **fix**: For user-visible bug fixes.
 3. **ref**: For features, refactors and bug fixes in internal operations.
```

**Review by jennmueng — CHANGES_REQUESTED** [review]
Thanks for this!

Also, the linter is failing because it's not expecting ES6 syntax in a `.js` file (only expects it in typescript files), can you add `dangerfile.js` to `.eslintignore`, and we'll be good to go!

File: dangerfile.js (hunk 2 above)
Comment (lucas-zimerman) [reply dangerfile.js]: sure, thanks for pointing that out @jennmueng.

File: dangerfile.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+function getChangelogDetails() {
+  return `
+<details>
+<summary><b>Instructions and example for changelog</b></summary>
+
+Please add an entry to \`CHANGELOG.md\` to the "Unreleased" section under the following heading:
+ 1. **feat**: For new user-visible functionality.
+ 2. **fix**: For user-visible bug fixes.
+ 3. **fef**: For features, refactors and bug fixes in internal operations.
```
Comment (jennmueng) [inline dangerfile.js:18]:
```suggestion
 3. **ref**: For features, refactors and bug fixes in internal operations.
```

**Review by jennmueng — APPROVED** [review]
One small thing, and otherwise LGTM

**Pull request merged**